plugins/catalog/src/api/CatalogClient.ts

```diff
@@ -16,6 +16,7 @@
 
 import { CatalogApi } from './types';
 import { DescriptorEnvelope } from '../types';
+import { Entity } from '@backstage/catalog-model';
```

Comment:
you can import `Location` from here too

Reply:
> OK but this is gonna be hammering the backend, I think it's better overall if we instead store the relevant data as an annotation to be taken directly from the entity instead of going for extra requests...

Yes, i agree. Fetching this after isn't very nice to the backend, and it makes the code more complex. We could create an API that resolves this (and possibly other) references when fetching the entities?

And yes, i've done a bad job with types so far here. :(